streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBRangeIteratorTest.java

```diff
@@ -0,0 +1,419 @@
+        rocksIterator.seek(key1Bytes.get());
+        expect(rocksIterator.isValid())
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(false);
+        expect(rocksIterator.key())
+            .andReturn(key1Bytes.get())
+            .andReturn(key2Bytes.get())
+            .andReturn(key3Bytes.get());
+        expect(rocksIterator.value()).andReturn(valueBytes).times(3);
+        rocksIterator.next();
+        expectLastCall().times(3);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+                storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key1Bytes,
+            key3Bytes,
+            true
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key1Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key2Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key3Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldReturnAllKeysInTheRangeReverseDirection() {
+
+        final RocksIterator rocksIterator = mock(RocksIterator.class);
+        rocksIterator.seekForPrev(key3Bytes.get());
+        expect(rocksIterator.isValid())
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(false);
+        expect(rocksIterator.key())
+            .andReturn(key3Bytes.get())
+            .andReturn(key2Bytes.get())
+            .andReturn(key1Bytes.get());
+        expect(rocksIterator.value()).andReturn(valueBytes).times(3);
+        rocksIterator.prev();
+        expectLastCall().times(3);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+                storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key1Bytes,
+            key3Bytes,
+            false
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key3Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key2Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key1Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldReturnAllKeysWhenLastKeyIsGreaterThanLargestKeyInStateStoreInForwardDirection() {
+
+        final String to = "f";
+        final  Bytes toBytes = Bytes.wrap(to.getBytes());
```

Comment:
```suggestion
        final Bytes toBytes = Bytes.wrap(to.getBytes());
```
Please also check other occurrences of this additional space.